src/executables/sysrepo-plugind.c

```diff
@@ -588,6 +653,11 @@ main(int argc, char **argv)
     rc = EXIT_SUCCESS;
 
 cleanup:
+    if (pidfile) {
```

Comment:
I appreciate the intention to unify all the conditions but unfortunately here it will not work, `pidfd` may easily be `-1` even though `pidfile` was set, so please revert this back.

Reply:
Doh. Not sure what I was thinking there. Thanks for spotting!